engine/runtime-compiler/src/main/java/org/enso/compiler/docs/DocsGenerate.java

```diff
@@ -32,28 +33,53 @@ public static <File> File write(
       DocsVisit visitor, org.enso.pkg.Package<File> pkg, Iterable<CompilerContext.Module> modules)
       throws IOException {
     var fs = pkg.fileSystem();
-    var docs = fs.getChild(pkg.root(), "docs");
-    var api = fs.getChild(docs, "api");
-    fs.createDirectories(api);
+    var apiDir = defaultOutputDir(pkg);
+    fs.createDirectories(apiDir);
 
     for (var module : modules) {
+      if (module.isSynthetic()) {
+        continue;
+      }
       var ir = module.getIr();
       assert ir != null : "need IR for " + module;
       if (ir.isPrivate()) {
         continue;
       }
       var moduleName = module.getName();
-      var dir = createPkg(fs, api, moduleName);
+      var dir = createDirs(fs, apiDir, stripNamespace(moduleName));
       var md = fs.getChild(dir, moduleName.item() + ".md");
       try (var mdWriter = fs.newBufferedWriter(md);
           var pw = new PrintWriter(mdWriter)) {
         visitModule(visitor, moduleName, ir, pw);
       }
     }
+    return apiDir;
+  }
+
+  public static <File> File defaultOutputDir(org.enso.pkg.Package<File> pkg) {
+    var fs = pkg.fileSystem();
+    var docs = fs.getChild(pkg.root(), "docs");
+    var api = fs.getChild(docs, "api");
```

Comment:
Maybe we could select the directory based on the _"type of documentation"_ - e.g. `api` for `--docs=api` and `md` for `--docs` or `--docs=md`